src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+#include "storage/deprecated_dictionary_column.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/run_length_column.hpp"
+#include "storage/value_column.hpp"
+
+namespace opossum {
+
+// select how many ranges we want in the filter
+// make this customizable?
+static constexpr uint32_t MAX_RANGES_COUNT = 10;
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
+ public:
+  virtual ~BaseFilter() = default;
+
+  /**
+   * checks whether the filter is able to determine that the given value 
+   * and predicate condition will not yield any positive results with the values
+   * represented by the filter data.
+   * 
+   * In other words: A scan operation with value and predicate_type on the column/chunk
+   * that this filter was created on would yield zero result rows.
+  */
+  virtual bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const = 0;
+};
+
+class ChunkColumnStatistics {
```

Comment:
Either make this final or add a virtual destructor to avoid possible misuse leading to memory leaks.

Reply:
Could use a short class comment as well